sympy/series/approximants.py

```diff
@@ -101,3 +110,289 @@ def approximants(l, X=Symbol('x'), simplify=False):
+            return numerator, denominator
+
+        numerator, denominator = next_numerator, next_denominator
+
+    # if it hasnt returned yet, then numerator.degree() == m
+    return numerator, denominator
+
+
+@public
+def pade_approximant(
+    f: Expr, x: Expr, x0: Expr | complex = 0, m: int=6, n: int | None = None
+) -> Expr:
+    """
+    `[m/n]` pade approximant of `f` around `x=0`.
+
+    Description
+    ===========
+
+    For a function `f` and integers `m` and `n`, the `[m/n]` pade approximant
+    of `f` is the rational function `p(x)/q(x)`, where `p(x)` and `q(x)` are
+    polynomials of degree at most `m` and `n` respectively. The pade
+    approximation, when it exists, is such that the taylor series of
+    `p(x)/q(x)` around `x=0` matches the taylor series of `f(x)` up to at
+    least order `(m + n)` in `x`.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The function to approximate.
+    x : Expr
+        The variable of the function.
+    x0 : Expr
+        The value around which the approximant is calculated. Defaults to 0.
+    m : int
+        The degree of the numerator polynomial. Defaults to 6.
+    n : int | None
+        The degree of the denominator polynomial. If None, `n = m`.
+
+    Returns
+    =======
+
+    pade approximant : Expr
+        returns `p(x)/q(x)`, where `p(x)/q(x)` is the pade approximant of `f`.
+
+    Examples
+    ========
+
+    >>> import sympy as sp
+    >>> from sympy.series.approximants import pade_approximant
+    >>> x = sp.symbols('x')
+
+    >>> pade_exp = pade_approximant(sp.exp(x), x, 0, 2)
+    >>> pade_exp
+    (x**2/4 + 3*x/2 + 3)/(x**2/4 - 3*x/2 + 3)
+
+    The numerators and denominators of an `[m/n]` pade approximant do not
+    necessarily have order `m` and `n` respectively.
+
+    >>> pade_sin = pade_approximant(sp.sin(x), x, 0, 1, 3)
+    >>> pade_sin
+    36*x/(6*x**2 + 36)
+
+    `[1/1]` Pade approximant of `log(x)` around `x0=1`
+
+    >>> pade_log = pade_approximant(sp.log(x), x, 1, 1)
+    >>> pade_log
+    4*(x - 1)/(2*x + 2)
+
+    The `[m/0]` pade approximant is the mth order taylor polynomial of `f`
+
+    >>> pade_cos = pade_approximant(sp.cos(x), x, 0, 4, 0)
+    >>> pade_cos
+    x**4/24 - x**2/2 + 1
+
+    The pade approximant does not always exist, for example, the `[1/1]`
+    approximant for `cos(x)`
+
+    >>> pade_cos11 = pade_approximant(sp.cos(x), x, 0, 1)
+    >>> pade_cos11
+    1
+
+    `1` does not match the taylor expansion of `cos(x)` up to second order in
+    `x`.
+
+    See also
+    ========
+
+    sympy.series.approximants.approximants
+    sympy.series.approximants.pade_approximant_gcdex
+    sympy.series.approximants.pade_approximants_gcdex
+    """
+    if n is None:
+        n = m
+
+    if m < 0 or n < 0:
+        raise ValueError(f"{m=} and {n=} must be non-negative integers")
+
+    f_taylor_poly = f.series(x, x0, n=m + n + 1).subs(x, x + x0)\
+                    .removeO().as_poly(x, field=True)
```

Comment:
I think that the previous code here was better:
```python
f.subs(x, x + x0).series(x, 0, ...).removeO().as_poly(x)
```
Internally if `x0 != 0` then series just does the same substitution and back so this ends up doing three substitutions rather than one.